src/agents/tool_context.py

```diff
@@ -8,16 +8,23 @@ def _assert_must_pass_tool_call_id() -> str:
     raise ValueError("tool_call_id must be passed to ToolContext")
 
 
+def _assert_must_pass_tool_name() -> str:
+    raise ValueError("tool_name must be passed to ToolContext")
+
+
 @dataclass
 class ToolContext(RunContextWrapper[TContext]):
     """The context of a tool call."""
 
+    tool_name: str = field(default_factory=_assert_must_pass_tool_name)
+    """The name of the tool being invoked."""
+
     tool_call_id: str = field(default_factory=_assert_must_pass_tool_call_id)
     """The ID of the tool call."""
 
     @classmethod
     def from_agent_context(
-        cls, context: RunContextWrapper[TContext], tool_call_id: str
+        cls, context: RunContextWrapper[TContext], tool_name: str, tool_call_id: str
```

Comment:
Most developers never use this method, but since this is not a private method naming, so this could be a breaking change. so, keep tool_call_id as the second arg, then add 3rd arg `tool_call` as optional. At this moment, we may add only the `tool_name` but we may add more properties from `tool_call` in the future.